src/issues/currentIssue.ts

```diff
@@ -201,13 +201,19 @@ export class CurrentIssue {
 		}
 		const state: IssueState = this.stateManager.getSavedIssueState(this.issueModel.number);
 		this._branchName = this.shouldPromptForBranch ? undefined : state.branch;
+		const branchNameConfig = await variableSubstitution(
+			await this.ensureBranchTitleConfigMigrated(),
+			this.issue,
+			undefined,
+			await this.getUser(),
+		);
+		if (this._branchName !== branchNameConfig) {
+			const branchExists = await this.branchExists(this._branchName!);
+			if (!branchExists) {
+				this._branchName = branchNameConfig;
+			}
+		}
 		if (!this._branchName) {
```

Comment:
We also need to do this `if` if `this._branchName` changed in the previous block since it includes branch name validation.

Reply:
@burkeholland this is the only change I see that is needed before merging!

Reply:
I was thinking something like this, since we still needed to validate the branch name:

https://github.com/microsoft/vscode-pull-request-github/pull/2909/files#diff-e0d767b263cbb178ca6cf7902e1934d49d08810ea2f4c26d97d23b98e5643210R231-R236